### PR TITLE
Vundle style plugins (external plugins)

### PR DESCRIPTION
**Pull request by bpinto**
- [x] Update documentation.
- [x] Upgrade guide.
- [x] Support update.
- [x] Add documentation to the code.
- [ ] Add tests.

### CODE REVIEW

File: functions/fish.fish [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# NAME
+#   fish - Extend default fish binary
+#
+# DESCRIPTION
+#   Extend fish binary to support plugins and themes installation
+#
+function fish -d "Extend fish binary"
```
Comment (rominf) [inline functions/fish.fish]: I don't like this. How can I run `install` file with `fish`?
```
fish.packages --install
```
looks quite good for me.

Comment (bobthecow) [inline functions/fish.fish]: Agreed. I'm not a fan of changing what `fish` means :)

Comment (bobthecow) [inline functions/fish.fish]: Why not have an `omf` binary to handle all the oh-my-fish things?

**Comment by bobthecow** [comment]
We could remove the dependency on git by using github's tarball downloads. Not sure if that's worth it, though.

**Comment by bobthecow** [comment]
Install and update seem to depend on `$fish_theme` and `$fish_plugins` being set, but those are deprecated, right?

File: functions/fish.fish (hunk 1 above)
Comment (bpinto) [inline functions/fish.fish]: Okay!

**Comment by bpinto** [comment]
We could remove git dependency, although I like how it makes things simple. I think I'm not opposed to supporting tarballs, just don't think there are more interesting features to add first.

`$fish_theme` and `$fish_plugins` and deprecated to be set manually but we use them by install and update so that the system continues working the same, meaning we don't break any functionality when someone upgrades omf.

**Pull request merged**